crux-llvm/test-data/golden/T972-fail.at-least-clang12.c

```diff
@@ -0,0 +1,18 @@
+// Adapted from the Clang test suite:
```

Comment:
Is there a reason not to use an `.at-least-clang9` suffix here? That way, it would be tested with all supported versions of Clang. (I'm hoping that the error message would be the same in each version.)

Reply:
I assumed that this was the set of all possible suffixes, is that not true?

https://github.com/GaloisInc/crucible/blob/b558ba2341b11c88e668ed3f772f4d1960275229/crux-llvm/test/Test.hs#L40

Reply:
Ah, I was under the mistaken impression that we were doing something more thorough than that. In that case, disregard my comment above. We might want to make these suffixes more flexible in the future, but it's not terribly important for this patch.

Reply:
The `tasty-sugar` matching process attempts to match the parameter value if present; if it's not present it will match against the non-parameterized version of the filename.  

In this particular case, if you expect this to compile and build with all versions of clang, simply remove any `clang-range` value from the filename.

For more info, see https://hackage.haskell.org/package/tasty-sugar-1.1.1.0/docs/Test-Tasty-Sugar.html#t:CUBE and https://github.com/kquick/tasty-sugar/tree/master/examples/params

Reply:
The thing is, this shouldn't build with all versions of `clang`, but rather only with Clang 9 and up. If I understand how `tasty-sugar` works correctly, that would require adding a new entry to `"clang-range"` for every version of Clang that we wanted to conditionally test. Certainly doable, although it does make me wonder if there's a way we can have only two `"pre-clang*"` and `at-least-clang*"` wildcard patterns that take care of this legwork once and for all.

Reply:
Well, we'd only need to add entries for inflection points we cared about.

Alternatively, we could define the `"clang-range"` parameter as a `Nothing`, which means "accept any value".  Then we could decide on some syntax we wanted to allow the tests to make the right determination.  Something like "preN", "postN", "N", or no field.  When defined as `Nothing`, tasty-sugar can't do any validation of the entries, and you can only treat one of your parameters as a value wildcard like this, but it's certainly doable.